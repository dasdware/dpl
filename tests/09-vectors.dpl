type Point := [ x: Number, y: Number ];

function add(a: Point, b: Point)
    := [ x := a.x + b.x, y := a.y + b.y ];

function print(p: Point)
    := print("[x: " + p.x.toString() + ", y: " + p.y.toString() + "]");

var y := 2;
var p1 := [ x := 1, y ];
var p2: Point := [ y := 2*3, x := 1 ];
print("p1     : "); print(p1); print("\n");
print("p2     : "); print(p2); print("\n");
print("p1 + p2: "); print(p1 + p2); print("\n");

p1 := [ ..p1, x := p1.x * 3 ];
p2 := [ ..p2, y := p2.x * 21 ];
print("p1     : "); print(p1); print("\n");
print("p2     : "); print(p2); print("\n");
print("p1 + p2: "); print(p1 + p2); print("\n");
