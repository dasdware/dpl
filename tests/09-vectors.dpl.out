p1     : [x: 1, y: 2]
p2     : [x: 1, y: 6]
p1 + p2: [x: 2, y: 8]
p1     : [x: 3, y: 2]
p2     : [x: 1, y: 21]
p1 + p2: [x: 4, y: 23]
